server/src/services/course.ts

```diff
@@ -31,22 +33,35 @@ export async function findCourseById(courseId: number): Promise<CourseWithTransl
   return course as CourseWithTranslation;
 }
 
-export async function findAllInstances(courseId: number): Promise<Array<CourseInstance>> {
-  const course: Array<Course> | null = await models.Course.findAll({
-    attributes: ['id'],
-    where: {
-      id: courseId
-    }
-  });
+export interface InstanceWithTeacher {
+  CourseInstance: CourseInstance
+  Teacher: User | null
+}
+
+export async function findAllInstances(courseId: number): Promise<Array<Course | Array<InstanceWithTeacher>>> {
```

Comment:
Would it be better to return a tuple here? For example:
```JavaScript
Promise<[Course, Array<InstanceWithTeacher>]>
```

Current one seems to imply the course data is also an array or we are repeating course data.

This tuple would be easy to destructure in the route function:
```JavaScript
const [ course, instances ]: [Course, Array<InstanceWithTeacher>] = await courseService.findAllInstances(courseId);
```